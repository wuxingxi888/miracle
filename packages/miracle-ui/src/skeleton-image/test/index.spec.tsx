import { mount } from '../../../test';
import { SkeletonImage } from '..';

test('should skeleton image render correctly', () => {
  const wrapper = mount(SkeletonImage);

  expect(wrapper.html()).toMatchSnapshot();
});

test('should skeleton image works with imageSize prop', () => {
  const wrapper = mount(SkeletonImage, {
    props: {
      imageSize: '20rem',
    },
  });
  const dom = wrapper.find('.mi-skeleton-image');

  expect(dom.style.width).toBe('20rem');
  expect(dom.style.height).toBe('20rem');
});

test('should skeleton image works with imageShape prop', () => {
  const wrapper = mount(SkeletonImage, {
    props: {
      imageShape: 'round',
    },
  });

  expect(wrapper.find('.mi-skeleton-image--round')).toBeTruthy();
});
