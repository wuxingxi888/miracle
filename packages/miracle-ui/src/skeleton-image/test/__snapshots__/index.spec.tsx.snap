// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should skeleton image render correctly 1`] = `
<div class="mi-skeleton-image mi-skeleton-image--square">
  <i class="mi-badge__wrapper mi-icon mi-icon-photo mi-skeleton-image__icon">
  </i>
</div>
`;
